api/core/text2sql_common.py

```
@@ -0,0 +1,188 @@
+    Applies validation identical to the original ``_graph_name`` in
+    ``text2sql.py``: strip, truncate to 200 chars, reject empty, bypass
+    prefix for general/demo graphs.
+
+    Raises:
+        GraphNotFoundError: If *graph_id* is empty after stripping.
+    """
+    graph_id = graph_id.strip()[:200]
+    if not graph_id:
+        raise GraphNotFoundError(
+            "Invalid graph_id, must be less than 200 characters."
+        )
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)